src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets

```diff
@@ -87,6 +87,9 @@ Copyright (c) .NET Foundation. All rights reserved.
   <!-- Checks for deprecated Aspire workload usage -->
   <Import Project="$(MSBuildThisFileDirectory)Microsoft.NET.AspireWorkloadDeprecation.targets" />
 
+  <!-- Checks for deprecated RoslynCompilerType usage -->
+  <Import Project="$(MSBuildThisFileDirectory)Microsoft.NET.RoslynCompilerTypeDeprecation.targets" />
```

Comment:
Let's not have a separate targets file for this; just inline the Target somewhere near the place where we compute RoslynCompilerType today